File: src/main/java/ee/carlrobert/codegpt/actions/IncludeFilesInContextNotifier.java
package ee.carlrobert.codegpt.actions;

import com.intellij.openapi.vfs.VirtualFile;
import com.intellij.util.messages.Topic;
import java.util.List;

public interface IncludeFilesInContextNotifier {

  Topic<IncludeFilesInContextNotifier> FILES_INCLUDED_IN_CONTEXT_TOPIC =
      Topic.create("filesIncludedInContext", IncludeFilesInContextNotifier.class);

  void filesIncluded(List<VirtualFile> includedFiles);
}

File: src/main/kotlin/ee/carlrobert/codegpt/actions/IncludeFilesInContextNotifier.kt
package ee.carlrobert.codegpt.actions

import com.intellij.util.messages.Topic
import ee.carlrobert.codegpt.ReferencedFile

fun interface IncludeFilesInContextNotifier {
  fun filesIncluded(includedFiles: List<ReferencedFile>)

  companion object {
    @JvmField
    val FILES_INCLUDED_IN_CONTEXT_TOPIC: Topic<IncludeFilesInContextNotifier> =
      Topic.create("filesIncludedInContext", IncludeFilesInContextNotifier::class.java)
  }
}
